### PR TITLE
Use both the NAD and the CRD in endpointslice mirror tests 

### PR DESCRIPTION
**Pull request by kyrtapz**
/cc @tssurya @ricky-rav

### CODE REVIEW

**Comment by maiqueb** [comment]
/cc @ormergi 

Could you take a look since you're more familiar with this `DescribeTableSubtree` container ?

File: test/e2e/network_segmentation_endpointslices_mirror.go [hunk 1]
```diff
@@ -44,186 +44,204 @@ var _ = Describe("Network Segmentation EndpointSlices mirroring", func() {
...
-					metav1.CreateOptions{},
+		DescribeTableSubtree("created using",
+			func(createNetworkFn func(c networkAttachmentConfigParams) error) {
+				DescribeTable(
+					"mirrors EndpointSlices managed by the default controller for namespaces with user defined primary networks",
+					func(
+						netConfig networkAttachmentConfigParams,
+						isHostNetwork bool,
+					) {
+						netConfig.namespace = f.Namespace.Name
```
Comment (ormergi) [inline test/e2e/network_segmentation_endpointslices_mirror.go:55]: Namespace is set at line 58, line seem redundant

File: test/e2e/network_segmentation_endpointslices_mirror.go [hunk 2]
```diff
@@ -44,186 +44,204 @@ var _ = Describe("Network Segmentation EndpointSlices mirroring", func() {
...
-
-				netConfig.namespace = f.Namespace.Name
-
-				By("creating the attachment configuration")
-				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
-					context.Background(),
-					generateNAD(netConfig),
-					metav1.CreateOptions{},
+		DescribeTableSubtree("created using",
+			func(createNetworkFn func(c networkAttachmentConfigParams) error) {
```
Comment (ormergi) [inline test/e2e/network_segmentation_endpointslices_mirror.go:173]: It seems `createNetworkFn` is not called in this tests

**Review by ormergi — COMMENTED** [review]
Hi Patryk, please see my inline comments

**Review by maiqueb — DISMISSED** [review]
Looks good.

File: test/e2e/network_segmentation_endpointslices_mirror.go (hunk 2 above)
Comment (kyrtapz) [reply test/e2e/network_segmentation_endpointslices_mirror.go:173]: good catch, copy-paste error
https://github.com/ovn-org/ovn-kubernetes/pull/4643/commits/4a2d447a68233d152d7b6971fc0b93363b69bdb4#diff-efa6082185cdcbcd4b0ef97df7abbe8ac5013a23b8530d5714e47845c4a272a1R179
Fixed.

**Comment by kyrtapz** [comment]
known flake: https://github.com/ovn-org/ovn-kubernetes/issues/4144#issuecomment-2301381260

**Review by maiqueb — APPROVED** [review]
Thank you.

**Review by ormergi — APPROVED** [review]
Thanks, LGTM

**Pull request merged**